Terminal.Gui/Views/Scroll.cs

```diff
@@ -0,0 +1,408 @@
+﻿namespace Terminal.Gui;
+
+/// <summary>
+///     Provides a proportional control for scrolling through content. Used within a <see cref="ScrollBar"/>.
+/// </summary>
+public class Scroll : View
+{
+    /// <inheritdoc/>
+    public Scroll ()
+    {
+        WantContinuousButtonPressed = true;
+        CanFocus = false;
+        Orientation = Orientation.Vertical;
+        Width = Dim.Auto (DimAutoStyle.Content, 1);
+        Height = Dim.Auto (DimAutoStyle.Content, 1);
+
+        _slider = new ()
+        {
+            Id = "slider",
+            Width = Dim.Auto (DimAutoStyle.Content),
+            Height = Dim.Auto (DimAutoStyle.Content),
+            WantMousePositionReports = true
+        };
+        Add (_slider);
+
+        Added += Scroll_Added;
+        Removed += Scroll_Removed;
+        Initialized += Scroll_Initialized;
+        MouseEvent += Scroll_MouseEvent;
+        _slider.MouseEvent += Slider_MouseEvent;
+        _slider.MouseEnter += Slider_MouseEnter;
+        _slider.MouseLeave += Slider_MouseLeave;
+    }
+
+    private readonly View _slider;
+
+    private int _lastLocation = -1;
+
+    private bool _wasSliderMouse;
+
+    private int _barSize => Orientation == Orientation.Vertical ? GetContentSize ().Height : GetContentSize ().Width;
```

Comment:
This doesn't quite do what you were thinking, because this won't cache.

This actually is still a property, and created an _additional_ copy and layer of indirection.

When a member is declared with `=>`, it is a get-only calculated property.

If you want a get-once-and-cache behavior for a property, this is how you declare it:

```csharp
private T _propertyName { get; } = expression;
```

But, that's not going to have identical behavior, because it will never evaluate `expression` again.

That's why I ultimately had just suggested the easiest thing for you for now would be to basically paste in that code snippet from one of the later comments.

Reply:
Done see https://github.com/gui-cs/Terminal.Gui/pull/3498/commits/5d2120eac69f2bec85947842b87d78daaa5b26c3.